### PR TITLE
Layered Architecture Task 1 Review

### PR DESCRIPTION
**Pull request by rkosmirak**

### CODE REVIEW

File: src/CartingService.ComponentTests/Cart/CartTests.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 namespace CartingService.ComponentTests.Cart;
 
 
-
+//todo did you heard anything about testing pyramid?
```
Comment (Yariki) [inline src/CartingService.ComponentTests/Cart/CartTests.cs:19]: Actually, nope, I need to check.

File: src/CartingService.UnitTests/Domain/CartTest.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@ namespace CartingService.UnitTests.Domain;
 
 public class CartTest
 {
+    //todo is this possible for this test to fail?
```
Comment (Yariki) [inline src/CartingService.UnitTests/Domain/CartTest.cs:8]: In this test, I check that when we create a Cart, CartItems always should be initialized. So, it shouldn't fail.

File: src/CartingService/Application/Cart/Commands/AddCartItem/AddItemCommand.cs [hunk 3]
```diff
@@ -33,10 +33,13 @@ public Task<bool> Handle(AddItemCommand request, CancellationToken cancellationT
         var cart = _cartRepository.GetCart(request.CartId);
         if (cart == null)
         {
+            //todo how does exceptions are handled
```
Comment (Yariki) [inline src/CartingService/Application/Cart/Commands/AddCartItem/AddItemCommand.cs:36]: good catch, need to add filter or middleware.

File: src/CartingService/Application/Cart/Commands/AddCartItem/AddItemCommand.cs [hunk 4]
```diff
@@ -33,10 +33,13 @@ public Task<bool> Handle(AddItemCommand request, CancellationToken cancellationT
...
             throw new CartServiceException("The cart does not exist");
         }
         
         var newItem = _mapper.Map<CartItem>(request.Item);
+        
+        //todo this looks like business logic, shouldn't it be in the Domain layer instead?
```
Comment (Yariki) [inline src/CartingService/Application/Cart/Commands/AddCartItem/AddItemCommand.cs:42]: Agree, In the first implementation I put it in the domain. But I do not remember the reason, why I moved here.

File: src/CartingService.UnitTests/Domain/CartTest.cs (hunk 2 above)
Comment (rkosmirak) [reply src/CartingService.UnitTests/Domain/CartTest.cs:8]: oh, i see, my mistake

**Pull request merged**